ci/flaky_test/process_xml.py

```diff
@@ -103,8 +103,17 @@ def parse_and_print_test_suite_error(testsuite, log_path):
             # finding the last test case to run. The expected format of that is:
             #     "[ RUN      ] <TestParams>/<TestSuite>.<TestCase>\n".
             last_test_fullname = test_output.split('[ RUN      ]')[-1].splitlines()[0]
-            last_testsuite = last_test_fullname.split('/')[1].split('.')[0]
-            last_testcase = last_test_fullname.split('.')[1]
+            last_test_fullname_splitted_on_slash = last_test_fullname.split('/')
```

Comment:
IIRC the `<TestParams>/` portion of the run name is optional.

I think you should split by '.' first and then optionally split on / if present.  Something like:
```
            last_test_fullname_splitted_on_dot = last_test_fullname.split('.')
            if len(last_test_fullname_splitted_on_dot) == 2:
                last_testcase = last_test_fullname_splitted_on_dot[1]
                testsuite_split_on_slash = 
                last_testsuite = last_test_fullname_splitted_on_dot[0].split('/')[-1]
            else:
                last_testcase = "Could not retrieve last test case"
                last_testsuite = "Could not retrieve last test suite"
```

Reply:
Is this format enforced? I thought we were parsing a string so I was trying to be as defensive as possible with my assumptions

Reply:
I was just pointing out that your version didn't correctly extract test names if the test is not parametrized.  I do appreciate being as defensive as possible.

Thanks for fixing this name extraction logic.